packages/nominal-types/README.md

```diff
@@ -0,0 +1,66 @@
+[![npm][npm-image]][npm-url]
+[![npm-downloads][npm-downloads-image]][npm-url]
+<br />
+[![code-style-prettier][code-style-prettier-image]][code-style-prettier-url]
+
+[code-style-prettier-image]: https://img.shields.io/badge/code_style-prettier-ff69b4.svg?style=flat-square
+[code-style-prettier-url]: https://github.com/prettier/prettier
+[npm-downloads-image]: https://img.shields.io/npm/dm/@solana/nominal-types?style=flat
+[npm-image]: https://img.shields.io/npm/v/@solana/nominal-types?style=flat
+[npm-url]: https://www.npmjs.com/package/@solana/nominal-types
+
+# @solana/nominal-types
+
+This package contains type utilities for creating nominal types in TypeScript.
+
+## Brands make equal values distinct at the type level
+
+```ts
+const unverifiedName = 'Alice';
+const verifiedName = unverifiedName as Brand<'Alice', 'VerifiedName'>;
+
+'Alice' satisfies Brand<string, 'VerifiedName'>; // ERROR
+unverifiedName satisfies Brand<string, 'VerifiedName'>; // ERROR
+verifiedName satisfies Brand<'Bob', 'VerifiedName'>; // ERROR
+verifiedName satisfies Brand<'Alice', 'VerifiedName'>; // OK
+verifiedName satisfies Brand<string, 'VerifiedName'>; // OK
+```
+
+## Values can be tagged as compressed data
+
+```ts
+const untaggedData = new Uint8Array([/ ... *\/]);
+const compressedData = untaggedData as CompressedData<typeof untaggedData, 'zstd'>;
+
+compressedData satisfies CompressedData<Uint8Array, 'zstd'>; // OK
+untaggedData satisfies CompressedData<Uint8Array, 'zstd'>; // ERROR
+```
+
+## Strings can be tagged as being encoded using a particular scheme
+
+```ts
+const untaggedString = 'dv1ZAGvdsz5hHLwWXsVnM94hWf1pjbKVau1QVkaMJ92';
+const encodedString = untaggedString as EncodedString<typeof untaggedString, 'base58'>;
```

Comment:
I like this! Solves the `Base58EncodedAddress` vs `Address` problem we had before